example/viewer.ts

```diff
@@ -105,6 +123,311 @@ export class Viewer {
+      geometry: geometry,
+      color: '#00FF00',
+      mesh: cube,
+      position: new THREE.Vector3(index, index, index),
+      rotation: new THREE.Quaternion,
+      scale: new THREE.Vector3(1, 1, 1),
+    });
+    this.scene.add(this.clipBoxes[index].mesh);
+  }
+
+  addClipSphere(clipSphere: IClipSphere): void {
+    const geometry = new THREE.SphereGeometry(1, 32, 32);
+    const material = new THREE.MeshLambertMaterial({ 
+      color: '#0000ff',
+      opacity: 0.25,
+      transparent: true, 
+      wireframe: true,
+    });
+    const sphere = new THREE.Mesh( geometry, material);
+    this.clipSpheres.push({
+      iClipSphere: clipSphere,
+      geometry: geometry,
+      color: '#0000FF',
+      mesh: sphere,
+      position: new THREE.Vector3(0, 0, 0),
+      rotation: new THREE.Quaternion,
+      scale: new THREE.Vector3(1, 1, 1),
+    });
+    this.scene.add(this.clipSpheres[0].mesh);
+  }
+
+  addClipPlane(clipPlane: IClipPlane): void {
+    const geometry = new THREE.PlaneGeometry( 5, 20, 32 );
+    const material = new THREE.MeshBasicMaterial({
+      color: '#ff00ff',
+      side: THREE.DoubleSide,
+      opacity: 0.25,
+      transparent: true, 
+    });
+    const plane = new THREE.Mesh( geometry, material );
+    this.clipPlanes.push({
+      iClipPlane: clipPlane,
+      geometry: geometry,
+      color: '#0000FF',
+      mesh: plane,
+      position: new THREE.Vector3(0, 0, 0),
+      rotation: new THREE.Quaternion(),
+      scale: new THREE.Vector3(1, 1, 1),
+    });
+    this.scene.add(plane);
+  }
+
+  addClipCylinder(clipCylinder: IClipCylinder): void {
+    const geometry = new THREE.CylinderGeometry( 1, 1, 1, 32 );
+    const material = new THREE.MeshBasicMaterial({
+      color: '#ff00ff',
+      wireframe: true,
+    });
+    const cylinder = new THREE.Mesh( geometry, material );
+    this.clipCylinders.push({
+      iClipCylinder: clipCylinder,
+      geometry: geometry,
+      color: '#0000FF',
+      mesh: cylinder,
+      position: new THREE.Vector3(0, 0, 0),
+      rotation: new THREE.Quaternion(),
+      scale: new THREE.Vector3(1, 1, 1),
+    });
+    this.scene.add(cylinder);
+  }
+
+  updateClipMode() {
+    const pcos = this.pointClouds;
+    const numPcos = pcos.length;
+    if (numPcos > 0) {
+      const pointCloud = pcos[0];
+      if (pointCloud.material.clipMode === ClipMode.HIGHLIGHT_INSIDE) {
+        pointCloud.material.clipMode = ClipMode.CLIP_OUTSIDE;
+      } else if (pointCloud.material.clipMode === ClipMode.CLIP_OUTSIDE) {
+        pointCloud.material.clipMode = ClipMode.CLIP_INSIDE;
+      } else if (pointCloud.material.clipMode === ClipMode.CLIP_INSIDE) {
+        pointCloud.material.clipMode = ClipMode.HIGHLIGHT_INSIDE;
+      }
+    }
+  }
+
+  translateClipBox = (translation: Vector3, axis: string) => {
+    const currentPos = this.clipBoxes[0].position;
+    if (axis === 'x') {
+      this.clipBoxes[0].position = new Vector3(translation.x, currentPos.y, currentPos.z);
```

Comment:
`this.clipBoxes[0].position` is the same vector as the one referenced by `currentPos` so you could do:

```ts
this.clipBoxes[0].position.x = translation.x;
```
But either way, it feels a bit strange that we are passing a full `Vector3` into this method, plus an axis. Why not just have the `x`, `y` and `z` translations or a vector that we `.add` to the current position vector?